graph/schema.resolvers.go

```diff
@@ -0,0 +1,213 @@
+package graph
+
+// This file will be automatically regenerated based on the schema, any resolver implementations
+// will be copied through when generating and any unknown code will be moved to the end.
+// Code generated by github.com/99designs/gqlgen version v0.17.43
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/TanyaEIEI/pokedex/graph/model"
+)
+
+// CreatePokemon is the resolver for the createPokemon field.
+func (r *mutationResolver) CreatePokemon(ctx context.Context, input model.CreatePokemon) (*model.Pokemon, error) {
+	context := GetContext(ctx)
+	var pokemonList []*model.Pokemon
+	err := context.Database.Find(&pokemonList).Error
+	if err != nil {
+		return nil, err
+	}
+	currentId, errConvertId := getCurrentId(pokemonList)
+	if errConvertId != nil {
+		return nil, err
+	}
+	newPokenmon := &model.Pokemon{
+		ID:          strconv.Itoa(currentId + 1),
+		Name:        input.Name,
+		Description: input.Description,
+		CategoryID:  input.CategoryID,
+		AbilityID:   input.AbilityID,
+		TypeID:      input.TypeID,
+	}
+	err2 := context.Database.Create(&newPokenmon).Error
```

Comment:
thus the variable `err` is not reused, You can use `err` variable instead of `err2`.